drpcconn/conn.go

```diff
@@ -66,17 +67,29 @@ func (c *Conn) Invoke(ctx context.Context, rpc string, in, out drpc.Message) (er
 	mon.Event("outgoing_requests")
 	mon.Event("outgoing_invokes")
 
-	data, err := proto.Marshal(in)
-	if err != nil {
-		return errs.Wrap(err)
-	}
-
 	stream, err := c.man.NewClientStream(ctx)
 	if err != nil {
 		return err
 	}
 	defer func() { err = errs.Combine(err, stream.Close()) }()
 
+	invokeMetadata := make([]byte, 0)
+	metadata, ok := drpcmetadata.Get(ctx)
+	if ok {
+		invokeMetadata, err = metadata.Encode(invokeMetadata)
+		if err != nil {
+			return err
+		}
+		if err := stream.RawWrite(drpcwire.KindInvokeMetadata, invokeMetadata); err != nil {
+			return err
+		}
+	}
+
+	data, err := proto.Marshal(in)
+	if err != nil {
+		return errs.Wrap(err)
+	}
```

Comment:
Can we move this back up to where it was? I prefer to handle this error before we interact with the manager. The idea is to get all of the locally erroring things done before potentially doing network erroring things.